subsys/bluetooth/mesh/Kconfig

```diff
@@ -103,6 +112,57 @@ menuconfig BT_MESH_ADV_EXT
 
 if BT_MESH_ADV_EXT
 
+choice BT_MESH_WORKQ_CONTEXT
```

Comment:
I still do not understand why `system queue` is required if there is a separate work queue for it.
I didn't get any meaningful explanation of it. What is use case for it?
It looks like unnecessary overcomplication to me, both implementation and maintenance.

Reply:
I'm thinks the bigger reason of keep sysworkq is less ram comsume.

Reply:
https://github.com/zephyrproject-rtos/zephyr/actions/runs/11288115630/job/31395274065?pr=79690

I have to say that using K_NO_WAIT is not as stable as before.

Reply:
I agree with your opinion. I have resubmitted it now and set Mesh Workq as the default, and added a description in the doc. 

1. For bsim, all tests will be based on Mesh Workq by default, and additional tests based on sysworkq will be added.

3. For resource overhead, I think it is not easy to use adv_ext.c for products with insufficient resources, such as microbit_v1.

Reply:
As for whether to retain sysworkq, I think we can refer to the implementation of hci core(https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/hci_core.c#L4072). Maybe we can remove it later, but I think there is no big problem in retaining it now.

Reply:
I agree that we need to keep sysworkq option for devices with the limited RAM. IF the mesh stack is the only user of the host, it is possible to estimate the required number of tx buffers, which will always work.

Reply:
I'm sorry, there was an issue with my previous BSIM environment. I will resolve it instead of bothering you.